packages/web-media-box/src/pipelines/mse/mseManager.ts

```diff
@@ -0,0 +1,178 @@
+import type Logger from "@/utils/logger";
+import type { SourceBufferOperation, SourceBufferWrapper } from "./types/bufferOperation";
+import { OperationType } from "./consts/sourceBuffer";
+
+export default class MseManager {
+  private readonly logger: Logger;
+  private mediaSource: MediaSource;
+  private sourceBuffers: Map<string, SourceBufferWrapper>;
+  private sourceOpen: Promise<void>;
+
+  public constructor(logger: Logger) {
+    this.logger = logger.createSubLogger('MseManager');
+    //TODO: ManagedMediaSource
+    this.mediaSource = new MediaSource();
+    this.sourceOpen = this.initMediaSource();
+    this.sourceBuffers = new Map();
+  }
+
+  public initBuffers(mimeCodecs: Array<string>) {
+    mimeCodecs.forEach((mimeCodec) => {
+      if (MediaSource.isTypeSupported(mimeCodec)) {
+        const mimeType = mimeCodec.substring(0, mimeCodec.indexOf(';'));
+        this.addSourceBuffer(mimeType);
+      } else {
+        this.logger.warn(`MediaSource cannot create SourceBuffer for type ${mimeCodec}`);
+      }
+    });
+  }
+
+  public appendData(mimeType: string, data: ArrayBuffer) {
+    const bufferWrapper = this.sourceBuffers.get(mimeType);
+    if (bufferWrapper) {
+      const buffer = bufferWrapper.buffer;
+      const queue = bufferWrapper.queue;
+      const operation = (): Promise<void> => {
+        return new Promise((resolve, reject) => {
+          const removeAllEventListeners = () => {
+            buffer.removeEventListener('updateend', updateEnd);
+            buffer.removeEventListener('error', onError);
+            buffer.removeEventListener('abort', onAbort);
+          };
+          // TODO: handle update/updatestart events
```

Comment:
not really sure we need them, maybe only for debug/logs purposes, but this will be noisy

Reply:
Yeah debug logging was the only rationality I could think of as well.